pkg/codec/element_extractor.go

```diff
@@ -9,6 +9,9 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/types"
 )
 
+// ElementExtractorLocation is used to determine which element to extract from a slice or array.
+// The default is ElementExtractorLocationMiddle, which will extract the middle element.
+// valid json values are "first", "middle", and "last".
```

Comment:
I'm confused by this whole setup.

Let's say you have a slice of 10 elements.  Which element gets extracted if you select "middle", and why is "middle" the default?  How would I extract, say, element 7?

Reply:
https://github.com/smartcontractkit/chainlink-common/pull/262#discussion_r1411196194

Reply:
https://github.com/smartcontractkit/chainlink-common/pull/262#discussion_r1411196194

Reply:
For now, we support first, middle, and last.  Middle was used for median and seemed to me to be the most likely one people would want.  I could not default, and simply throw.  I guess we could add index, but I'm not sure when it would be used or how easy it would be to use.  You may not always know the number of elements, my thought is it's mostly going to be based on the number of NOPs that reported.

Maybe I should change it to just be middle?  I didn't call it median, because it doesn't sort the data itself.

Reply:
I see... maybe just add a mention in the docstring that it will use the lower of the two closest to the middle if it's an even number of elements?  Not sure how to say that succinctly.

Reply:
If "middle" is the only current use case, I think leaving it as default is fine